draftlogs/7078_change.md

```diff
@@ -0,0 +1 @@
+ - Use `node.js v18` and `npm v10` in development and upgrade the version of `package-lock.json` to v3 [[#5922](https://github.com/plotly/plotly.js/pull/5922), [#7078](https://github.com/plotly/plotly.js/pull/7078)]
```

Comment:
This upgrade may considered a breaking change for some users e.g. node12, etc.
Do you think we really have to that at this moment?

Reply:
Yes, it's only been an hour since I bumped this, and I've already discovered a `Severity: high` vulnerability with mathjax that affect all users:

```sh
➜ npm audit
# npm audit report

elliptic  >=2.0.0
Elliptic's ECDSA missing check for whether leading bit of r and s is zero - https://github.com/advisories/GHSA-977x-g7h5-7qgw
Elliptic's EDDSA missing signature length check - https://github.com/advisories/GHSA-f7q4-pwc6-w24p
Elliptic allows BER-encoded signatures - https://github.com/advisories/GHSA-49q7-c7j4-3p7m
No fix available
node_modules/elliptic
  browserify-sign  >=3.0.2
  Depends on vulnerable versions of elliptic
  node_modules/browserify-sign
    crypto-browserify  >=3.11.0
    Depends on vulnerable versions of browserify-sign
    Depends on vulnerable versions of create-ecdh
    node_modules/crypto-browserify
      node-polyfill-webpack-plugin  *
      Depends on vulnerable versions of crypto-browserify
      node_modules/node-polyfill-webpack-plugin
  create-ecdh  >=2.0.1
  Depends on vulnerable versions of elliptic
  node_modules/create-ecdh

mathjax  <=2.7.9
Severity: high
MathJax Regular expression Denial of Service (ReDoS) - https://github.com/advisories/GHSA-v638-q856-grg8
fix available via `npm audit fix --force`
Will install mathjax@3.2.2, which is a breaking change
node_modules/mathjax-v2

6 vulnerabilities (5 low, 1 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
```

Reply:
The `mathjax v2` is used in our tests. 
Plotly.js also supports `mathjax v3`.

Reply:
Ah, you talk about the lockfile. I've reverted those changes.